crates/esthri-cli/src/http_server.rs

```diff
@@ -714,10 +714,50 @@ fn sanitize_filename(filename: String) -> String {
     )
 }
 
+async fn maybe_redirect<'a, T: S3 + Send>(
+    s3: &T,
+    bucket: &str,
+    path: &str,
+) -> Result<(bool, Option<Result<http::Response<Body>, warp::Rejection>>), warp::Rejection> {
+    let is_dir_listing = path.is_empty() || path.ends_with('/');
+    if !is_dir_listing && is_directory(s3, bucket, path).await? {
+        let path = String::from("/") + path + "/";
+        debug!("redirect for index-html: {}", path);
+        let response = Response::builder()
+            .header(LOCATION, path)
+            .status(StatusCode::FOUND)
+            .body(Body::empty())
+            .map_err(|err| EsthriRejection::warp_rejection(format!("{}", err)));
+        Ok((is_dir_listing, Some(response)))
+    } else {
+        Ok((is_dir_listing, None))
+    }
+}
+
+async fn maybe_serve_index_html<'a, T: S3 + Send>(
```

Comment:
Please add documentation. Role of function is not clear from name.

Reply:
done